src/structures/gdt.rs

```diff
@@ -61,28 +100,29 @@ impl GlobalDescriptorTable {
 }
 
 impl<const MAX: usize> GlobalDescriptorTable<MAX> {
-    /// Creates an empty GDT which can hold `MAX` number of [`Descriptor`]s.
+    /// Creates an empty GDT which can hold `MAX` number of [`Entry`]s.
     #[inline]
     pub const fn empty() -> Self {
         // TODO: Replace with compiler error when feature(generic_const_exprs) is stable.
         assert!(MAX > 0, "A GDT cannot have 0 entries");
         assert!(MAX <= (1 << 13), "A GDT can only have at most 2^13 entries");
+        const NULL: Entry = Entry::new(0);
         Self {
-            table: [0; MAX],
+            table: [NULL; MAX],
             len: 1,
         }
     }
 
-    /// Forms a GDT from a slice of `u64`.
+    /// Forms a GDT from a slice of raw [`Entry`] values.
     ///
     /// # Safety
     ///
     /// * The user must make sure that the entries are well formed
     /// * Panics if the provided slice has more than `MAX` entries
     #[inline]
-    pub const unsafe fn from_raw_slice(slice: &[u64]) -> Self {
+    pub const unsafe fn from_raw_entries(slice: &[u64]) -> Self {
```

Comment:
Slightly out of the scope for this pr: The documentation for this function mentions that one of the safety requirements is that the entries have to be well formed, yet we allow constructing `Descriptor`s with arbitrary content. If this is just about the null descriptor, we should probably just say that.

Reply:
Hmmm, that's a good point. We also can just check for the null descriptor at compile time.

 In that case, this method should probably be safe. I'll update the code to add the check.